p2p/src/net/mock/transport/tcp/adapter/noise.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use snowstorm::NoiseStream;
+use tokio::net::TcpStream;
+
+use crate::{error::P2pError, net::mock::peer::Role};
+
+use super::StreamAdapter;
+
+static NOISE_HANDSHAKE_PATTERN: &str = "Noise_XX_25519_ChaChaPoly_SHA256";
+
+static NOISE_HANDSHAKE_PARAMS: once_cell::sync::Lazy<snowstorm::NoiseParams> =
+    once_cell::sync::Lazy::new(|| NOISE_HANDSHAKE_PATTERN.parse().expect("valid pattern"));
+
+#[derive(Debug)]
+pub struct NoiseEncryptionAdapter {}
+
+pub struct StreamConfig {
+    local_key: snowstorm::Keypair,
+}
+
+impl Default for StreamConfig {
+    fn default() -> Self {
+        let local_key = snowstorm::Builder::new(NOISE_HANDSHAKE_PARAMS.clone())
+            .generate_keypair()
+            .expect("key generation must succeed");
+        Self { local_key }
+    }
+}
+
+#[async_trait]
+impl StreamAdapter for NoiseEncryptionAdapter {
+    type StreamConfig = StreamConfig;
+
+    type Stream = snowstorm::NoiseStream<TcpStream>;
+
+    async fn handshake(
+        stream_config: &Self::StreamConfig,
```

Comment:
Why not pass `&self` instead of this config, and load the key from it?

Reply:
I think I answered that above.